### ISSUE TITLE
Xpath locators with index won't work with 2.49.2

### ISSUE THREAD

**Issue by molsky** [SeleniumHQ/selenium#1524]
After upgrading to latest Selenium our Robot Framework tests with Selenium2Library failed when there were xpath locators with index selector: `TypeError: string indices must be integers`.

Wrote pure Python Selenium test:
```
driver = webdriver.Firefox()
driver.get(page_url)
driver.find_element_by_xpath("(//input[@value='Tutustu'])[13]").click()
```
And the test failed.
```
Traceback (most recent call last):
  File "test.py", line 6, in <module>
    driver.find_element_by_xpath("(//input[@value='Tutustu'])[13]").click()
  File "/Users/USER/robot_venv/venv/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 75, in click
    self._execute(Command.CLICK_ELEMENT)
  File "/Users/USER/robot_venv/venv/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 454, in _execute
    return self._parent.execute(command, params)
  File "/Users/USER/robot_venv/venv/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 201, in execute
    self.error_handler.check_response(response)
  File "/Users/USER/robot_venv/venv/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 107, in check_response
    message = value["value"]["message"]
TypeError: string indices must be integers
```
Selenium: 2.49.2
Python: 2.7.11
OS: OS X 10.10.5

**Comment by barancev** [comment]
Please retest in 2.50

**Comment by molsky** [comment]
This happens also with version 2.50.

**Comment by barancev** [comment]
We need a more robust proof :)

![01-02-2016 11-50-25](https://cloud.githubusercontent.com/assets/617090/12712548/2b88e9a0-c8da-11e5-9366-dcd757f05692.png)

**Comment by molsky** [comment]
```shell
(venv)[molsky selenium-py] (10:57:52) pip install -U selenium
Collecting selenium
  Using cached selenium-2.50.0-py2-none-any.whl
Installing collected packages: selenium
  Found existing installation: selenium 2.47.3
    Uninstalling selenium-2.47.3:
      Successfully uninstalled selenium-2.47.3
Successfully installed selenium-2.50.0
(venv)[molsky selenium-py] (10:57:56) python test.py 
  File "test.py", line 13, in <module>
    driver.find_element_by_xpath("(//input[@value='Tutustu'])[13]").click()
  File "/Users/molsky/robot_venv/venv/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 75, in click
    self._execute(Command.CLICK_ELEMENT)
  File "/Users/molsky/robot_venv/venv/lib/python2.7/site-packages/selenium/webdriver/remote/webelement.py", line 460, in _execute
    return self._parent.execute(command, params)
  File "/Users/molsky/robot_venv/venv/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 201, in execute
    self.error_handler.check_response(response)
  File "/Users/molsky/robot_venv/venv/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 107, in check_response
    message = value["value"]["message"]
TypeError: string indices must be integers
```
And here is the test script.
```python
from selenium import webdriver
from selenium.webdriver.common.keys import Keys

driver = webdriver.Firefox()

driver.get(url)
driver.find_element_by_xpath("(//input[@value='Tutustu'])[13]").click()
```

**Comment by barancev** [comment]
Ah, sorry, it's a click issue, I misinterpreted it to be an issue in find_element_... command.

So it must be unrelated to the locator type, instead we'll need to know what's the element you're trying to click and why it can't be clicked.

```
C:\Users\alexei>c:\Python27\python.exe
Python 2.7.11 (v2.7.11:6d1b6a68f775, Dec  5 2015, 20:32:19) [MSC v.1500 32 bit (
Intel)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> from selenium import webdriver
>>> driver = webdriver.Firefox()
>>> driver.get("http://www.seleniumhq.org/")
>>> driver.find_element_by_xpath("(//input)[1]").click()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "c:\Python27\lib\site-packages\selenium\webdriver\remote\webelement.py",
line 75, in click
    self._execute(Command.CLICK_ELEMENT)
  File "c:\Python27\lib\site-packages\selenium\webdriver\remote\webelement.py",
line 460, in _execute
    return self._parent.execute(command, params)
  File "c:\Python27\lib\site-packages\selenium\webdriver\remote\webdriver.py", l
ine 201, in execute
    self.error_handler.check_response(response)
  File "c:\Python27\lib\site-packages\selenium\webdriver\remote\errorhandler.py"
, line 188, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.ElementNotVisibleException: Message: Element is not c
urrently visible and so may not be interacted with
>>>
```

**Comment by molsky** [comment]
Yeah, tested some more and looks like that it is related to this another topic. There is fixed top-menu item and that hides those 'Tutustu' buttons when the page is scrolled to reach those ones that are not visible in current browser view.
https://github.com/SeleniumHQ/selenium/issues/1202#issuecomment-176209746

I can share complete test script via email that contains the page url where this can be reproduced if you are interested.

**Comment by barancev** [comment]
Error message should be fixed by commit c68ace0
